File: translation_output/misleading_test/wizardcoder/codenet/java/python-sanitized/p02256.py
import java.io.*;

class p02256:
    def main(self):
        br = BufferedReader(InputStreamReader(System.in))
        input = br.readLine().split(" ")
        a = int(input[0])
        b = int(input[1])

        System.out.println(self.GCD(a, b))

    def GCD(self, a, b):
        if a < b:
            temp = a
            a = b
            b = temp
        if a % b == 0:
            return b
        else:
            return self.GCD(b, a%b)

File: Results/DER/Translation/translation_output/w_test/deepseekcoder/codenet/python/java/p02256.java
<｜begin▁of▁sentence｜>You are an expert Python programmer.Your task is to write a Python function to solve a programming problem.

### Instruction:
Translate the following python code to java and enclose your solution inside ```java```.
A sample test case is provided below:

Test input:
54 20

Expected output:
2


```
x, y = map(int, input().split())
if x > y:
  x, y = y, x
while x % y != 0:
  x, y = y, x % y
print(y)

```

### Response:
```java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int x = scanner.nextInt();
        int y = scanner.nextInt();
        if (x > y) {
            int temp = x;
            x = y;
            y = temp;
        }
        while (x % y != 0) {
            int temp = y;
            y = x % y;
            x = temp;
        }
        System.out.println(y);
    }
}
```

This Java code does the same thing as the Python code. It reads two integers from the input, swaps them if necessary, and then calculates and prints the greatest common divisor (GCD) of the two numbers.
<|EOT|>
